models/DataObject/ClassDefinition/Data/Extension/RelationFilterConditionParser.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * @category    pimcore
+ * @date        25/06/2021
+ * @author      Filip Szenborn <fszenborn@divante.co>
+ */
+declare(strict_types=1);
+
+namespace Pimcore\Model\DataObject\ClassDefinition\Data\Extension;
+
+/**
+ * Trait RelationFilterConditionParser
+ * @package Pimcore\Model\DataObject\ClassDefinition\Data\Extension
+ */
+trait RelationFilterConditionParser
+{
+    /**
+     * Parses filter value of a relation field and creates the filter condition
+     * @param string $value
+     * @param string $operator
+     * @param string $name
+     * @return string
+     */
+    public function getRelationFilterCondition($value, $operator, $name)
+    {
+        if ($operator == '=') {
+            return '`' . $name . '` LIKE ' . "'%," . $value . ",%'";
+        }
+        $result = $name . " IS NULL";
+        $values = explode(',', (string)$value ?? '');
+        if (is_array($values) && !empty($values)) {
+            $fieldConditions = array_map(function ($value) use ($name) {
+                return '`' . $name . "` LIKE '%" . $value . "%' ";
```

Comment:
This returns wrong dataset, specially in case of ManyToOne relation. so when i select `manufacturer_id` = "4" it returns rows with ids 4,45,46 and so on...
![image](https://user-images.githubusercontent.com/5137917/135416836-77cf5940-511f-43bb-a6eb-ee382b6a7679.png)

Also, for ManyToManyRelation, the condition should be `(categories LIKE '%,558,%'  AND categories LIKE '%,559,%' )`

Reply:
I will check it out. I must have messed sth up while appling suggestions from jdevries14 review